app/src/main/java/io/homeassistant/companion/android/notifications/NotificationContentReceiver.kt

```diff
@@ -0,0 +1,47 @@
+package io.homeassistant.companion.android.notifications
+
+import android.app.NotificationManager
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import io.homeassistant.companion.android.util.UrlHandler
+import io.homeassistant.companion.android.util.cancelGroupIfNeeded
+import io.homeassistant.companion.android.webview.WebViewActivity
+
+class NotificationContentReceiver : BroadcastReceiver() {
+    companion object {
+        const val EXTRA_NOTIFICATION_GROUP = "EXTRA_NOTIFICATION_GROUP"
+        const val EXTRA_NOTIFICATION_GROUP_ID = "EXTRA_NOTIFICATION_GROUP_ID"
+        const val EXTRA_NOTIFICATION_ACTION = "EXTRA_NOTIFICATION_ACTION"
+    }
+
+    override fun onReceive(context: Context, intent: Intent) {
+
+        val action = intent.getStringExtra(EXTRA_NOTIFICATION_ACTION)
+        val group = intent.getStringExtra(EXTRA_NOTIFICATION_GROUP)
+        val groupId = intent.getIntExtra(EXTRA_NOTIFICATION_GROUP_ID, -1)
+
+        val notificationManager =
+            context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+
+        // Cancel any left empty group of the notification, if needed
+        // This is the case if the user clicked on the notification
+        // Then only the empty group is left and needs to be cancelled
+        notificationManager.cancelGroupIfNeeded(group, groupId)
+
+        if (!action.isNullOrBlank()) {
+            val intent = if (UrlHandler.isAbsoluteUrl(action)) {
+                Intent(Intent.ACTION_VIEW).apply {
+                    this.data = Uri.parse(action)
+                }
+            } else {
+                WebViewActivity.newInstance(context, action)
+            }
+
+            intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK
+
+            context.startActivity(intent)
```

Comment:
Isn't this code duplicated here: https://github.com/home-assistant/android/blob/e16bc38a7436f3c4d9eca93db9596e0451b29e67/app/src/main/java/io/homeassistant/companion/android/notifications/NotificationActionReceiver.kt#L95-L105

and dead code here: https://github.com/home-assistant/android/blob/e16bc38a7436f3c4d9eca93db9596e0451b29e67/app/src/main/java/io/homeassistant/companion/android/notifications/MessagingService.kt#L218-L237

Reply:
Yes it is a duplicate code. Didn't noticed that one. And also the dead code. I will remove it.